src/components/DataTable/DataTable.tsx

```diff
@@ -75,7 +75,12 @@ export class DataTable extends React.PureComponent<
     } = this;
     let collapsed = false;
     if (table && scrollContainer) {
-      collapsed = table.scrollWidth > scrollContainer.clientWidth;
+      const headerCells = table.querySelectorAll(
+        headerCell.selector,
+      ) as NodeListOf<HTMLElement>;
+      const fixedColumnWidth = headerCells[0].offsetWidth;
+      collapsed =
+        table.scrollWidth + fixedColumnWidth > scrollContainer.clientWidth;
```

Comment:
looks like we will need to add the `fixedColumnwidth` to this calculation as well.

Reply:
Since the CSS has a hardcoded value, for performance, I wonder if we aren't better using a token or hardcoded value for this. Thoughts?

Reply:
This was something I was looking to centralize when first implementing, but there wasn't a good way for us to share a single value between the SCSS and the JSX at the time. Now that we have `polaris-tokens` it's actually possible 🙌 Let's ship this quick fix (tophatting in iOS again now 👀) and I'll open separate PRs for adding and using a token for the `fixedColumnWidth`.